doc/src/manual/code-loading.md

```diff
@@ -0,0 +1,321 @@
+# Code Loading
+
+Julia has two mechanisms for loading code:
+
+1. **Code inclusion:** e.g. `include("source.jl")`. Inclusion allows you to split a single program across multiple source files. The expression `include("source.jl")` causes the contents of the file `source.jl` to be evaluated inside of the module where the `include` call occurs, much as if the text of `source.jl` were pasted into that file in place of the `include` call. If `include("source.jl")` is called multiple times, `source.jl` is evaluated multiple times. The included path, `source.jl`, is interpreted relative to the file where the `include` call occurs. This makes it simple to relocate a subtree of source files. In the REPL, included paths are interpreted relative to the current working directory, `pwd()`.
```

Comment:
In particular "much as if the text of `source.jl` were pasted into that file in place of the `include` call" is only valid if the `include` occurs at top level.

Reply:
I'll let someone else (maybe @vtjnash or @JeffBezanson?) write more about `include` since that's not really my area of expertise. But sure, more could be said about it, I just don't know what.